libdnf/utils/fs.cpp

```diff
@@ -0,0 +1,29 @@
+#include <sys/stat.h>
+
+#include <filesystem>
+
+#include "fs.hpp"
+
+
+namespace libdnf::utils::fs {
+
+
+void makedirs_for_file(std::string file_path) {
+    // get an absolute path that may not exist
+    auto path = std::filesystem::weakly_canonical(file_path);
+
+    // remove the file name
+    path = path.parent_path();
+
+    // iterate the absolute path from the beginning and create the missing parts
+    std::filesystem::path path_to_create;
+    for (auto & part : path) {
+        path_to_create /= part;
+        if (!std::filesystem::exists(path_to_create)) {
+            mkdir(path_to_create.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
```

Comment:
Return value of `mkdir` is ignored. Errors are ignored and not reported to the caller. But I see, in original code was the same problem. So, we can leave it as is.
Note (or TODO) about ignored errors?

Reply:
I've added TODO